Library/Formula/agda.rb

```diff
@@ -41,6 +42,10 @@ class Agda < Formula
     url "https://github.com/agda/agda-stdlib.git", :branch => "2.4.2.4"
   end
 
+  # fix compilation of the included Emacs mode
+  # to be removed once https://github.com/agda/agda/pull/1700 is merged
+  patch :DATA
```

Comment:
No. All that is conditional on `build.with? "emacs"` is running `agda-mode compile`. The user can run this command manually, having installed, say, a GUI version of Emacs, and the command will fail without the patch.